### PR TITLE
Add flake8 in pre-commit configuration

### PR DESCRIPTION
**Pull request by Pierre-Sassoulas**
## Description

Fix the legacy problem with flake8, some of them quite ancient (python 2.6 deprecation!) and add an automated checker in pre-commit configuration.

## Type of Changes

|     | Type                   |
| --- | ---------------------- |
| ✓   | :bug: Bug fix          |
| ✓   | :hammer: Refactoring   |

### CODE REVIEW

**Comment by Pierre-Sassoulas** [comment]
Ready, but need #972 to be merged first.

File: .flake8 [hunk 1]
```diff
@@ -0,0 +1,4 @@
+[flake8]
+extend-ignore = E203, E266, E501, C901, F401
```
Comment (cdce8p) [inline .flake8:2]:
```suggestion
extend-ignore = E203,E266,E501,C901,F401
```

Don't really know if that would make a difference

**Review by cdce8p — APPROVED** [review]
LGTM. One small comment

File: .flake8 (hunk 1 above)
Comment (Pierre-Sassoulas) [reply .flake8]: Yeah I think so, I had to make the max_complexity higher and C901 was supposed to be ignored.

**Pull request merged**